test/tools/isoltest.cpp

```diff
@@ -290,8 +294,31 @@ SyntaxTestStats SyntaxTestTool::processPath(
 
 }
 
+/// Ensures the terminal is set up properly.
+void setupTerminal()
```

Comment:
If the plan is to enable color output for solc at some point, then we might consider moving this to ``libdevcore/CommonIO.cpp``, as we did with ``DisableConsoleBuffering`` and ``readStandardInputChar``.

By the way ``DisableConsoleBuffering`` is not tested for Windows as well, so it would be good to verify that it works (so far out policy was not to care about Windows for ``isoltest``, until somebody tries to actually use it on Windows, which now seems to be the case ;)).

Reply:
Also: do we want to, resp. have to restore the console mode after exit?

Reply:
I am totally open for moving, but I wanted to preserve that for a later stage (to resume my actual topics soonish, but without reading too many raw ansi escape sequences on my screen) :)

Basic idea: get escape sequences fixed on Windows (e.g. get colors working), but provide a more generic solution once we are about to touch colorized diagnostics in solc post-0.5.0.

Reply:
no need to restore the mode, it's process-local, isn't it?

Reply:
I'm not sure - it may as well be "console-local" - that's why I'm asking :) (I'm not using all that much of Windows ;)).

And with moving I meant not changing anything in solc already, but just to place this routine into libdevcore already now (since the platform dependent DisableConsoleBuffer stuff is in there already anyways) - but it doesn't really matter too much.

So if you think that the console mode is process-local and doesn't have to be restored, then I'm fine with this - whether in libdevcore or directly in isoltest :).

Reply:
so I just checked [empirically] ``SetConsoleMode()`` on Windows is process-local, so no need to restore the values upon exit.

I had a quick look into libdevcore to find a good place to put it, however, I believe we should find a somewhat more ideal solution *post* 0.5.0, where we will be focusing on improved output (UX), and I believe that by that time, the API will change most likely again. I'd keep it here for the time being then`(**)`